dev/io.openliberty.microprofile.telemetry.1.0.internal_fat/fat/src/io/openliberty/microprofile/telemetry/internal_fat/apps/telemetry/OpenTelemetryServlet.java

```diff
@@ -0,0 +1,72 @@
+/*******************************************************************************
+ * Copyright (c) 2022 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+
+package io.openliberty.microprofile.telemetry.internal_fat.apps.telemetry;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+import org.junit.Test;
+
+import componenttest.app.FATServlet;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.api.trace.Span;
+import io.opentelemetry.api.trace.SpanContext;
+import io.opentelemetry.api.trace.TraceFlags;
+import io.opentelemetry.api.trace.TraceState;
+import io.opentelemetry.api.trace.Tracer;
+import io.opentelemetry.context.Context;
+import jakarta.annotation.PostConstruct;
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import jakarta.servlet.annotation.WebServlet;
+
+@WebServlet("/opentelemetry")
+@ApplicationScoped
+public class OpenTelemetryServlet extends FATServlet {
+
+    @Inject
+    OpenTelemetry openTelemetry;
+
+    Tracer tracer;
+
+    private static final String SPAN_NAME = "MySpanName";
+    private static final SpanContext spanContext = SpanContext.create(
+                                                                      "00000000000000000000000000000061",
+                                                                      "0000000000000061",
+                                                                      TraceFlags.getDefault(),
+                                                                      TraceState.getDefault());
+
+    @PostConstruct
+    public void postConstruct() {
+        this.tracer = openTelemetry.getTracer("instrumentation-test", "1.0.0");
+    }
+
+    @Test
+    void testSpanContextPropagationExplicitParent() {
+        Span span = tracer
+                        .spanBuilder(SPAN_NAME)
+                        .setParent(Context.root().with(Span.wrap(spanContext)))
+                        .startSpan();
+        assertEquals(span.getSpanContext(), spanContext);
+    }
+
+    @Test
+    public void tracerTest() {
+        assertNotNull(tracer);
+    }
+
+    @Test
+    public void testOpenTelemetryNotNull() {
+        assertNotNull(openTelemetry);
+    }
```

Comment:
`testOpenTelemetryNotNull` doesn't test anything since the test will throw an exception during `postConstruct` if `openTelemetry` is `null`.

Generally we want to try to cause an exception in the test method rather than during startup if we can.

I'm not entirely sure what `testSpanContextPropagationExplicitParent()` is really here for. It might be helpful to test that we can do _something_ with the `tracer`, but this test looks like it's trying to assert some other behaviour which wouldn't really belong in this test.

I would move all of this code into one test method so that the first time we call the `openTelemetry` bean is in the test method.

Reply:
I have updated this test now